.github/workflows/check-and-test.yml

```diff
@@ -0,0 +1,88 @@
+name: Pipeline
+
+on:
+  workflow_call:
+    inputs:
+      flag:
+        required: false
+        type: string
+        description: "Additional flag to pass to the build script (e.g. --workspace=dito)"
+
+jobs:
+  check-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .node-version
+          cache: npm
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Cache npm dependencies
+        id: cache-npm-deps
+        uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        if: steps.cache-npm-deps.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - name: Audit used licences
```

Comment:
I first wanted to complain that merging test and audit licence jobs would slow down the pipeline but looked it up and this step only takes 1 second whereas the whole duplicated checkout and installing of dependencies is what actually wasted resources. 

So good job removing this duplication!